xds/src/main/java/io/grpc/xds/XdsClient.java

```diff
@@ -389,50 +389,60 @@ ListenerUpdate build() {
     }
   }
 
+  /**
+   * Watcher interface for a single requested xDS resource.
+   */
+  private interface ResourceWatcher {
+
+    /**
+     * Called when the resource discovery RPC encounters some transient error.
+     */
+    void onError(Status error);
+
+    /**
+     * Called when the requested resource is not available.
+     *
+     * @param resourceName name of the resource requested in discovery request.
+     */
+    void onResourceDoesNotExist(String resourceName);
```

Comment:
Based on the code elsewhere, it looks like you don't wait for a DiscoveryResponse before calling this but you may call it if you found `resourceName` in the `absent*Resources` set. Is that semantic correct and agreeable to everyone?

Reply:
`absent*Resources` is the cache for resources that known to be absent. For example, if some watcher is already watching resource A (sent A in request), but A does not exist. We need to memorize this information. When another watcher is added to also watch A, we know immediately it doesn't exist. If we do not memorize that information, the second watcher will never get notified.